galsim/chromatic.py

```diff
@@ -1567,9 +1661,26 @@ def drawImage(self, bandpass, image=None, integrator='trapezoidal', **kwargs):
             jac, offset, _ = self._getTransformations(bandpass.red_limit)
             int_im = galsim.Transform(int_im, jac=jac, offset=offset, gsparams=self.gsparams)
             image = int_im.drawImage(image=image, **kwargs)
+            self._last_wcs = image.wcs
             return image
         else:
-            return ChromaticObject.drawImage(self, bandpass, image, integrator, **kwargs)
+            image = ChromaticObject.drawImage(self, bandpass, image, integrator, **kwargs)
+            self._last_wcs = image.wcs
+            return image
+
+    @property
+    def noise(self):
```

Comment:
I took the noise bit out of `GSObject.__init__` entirely, and also removed the ability to pass in a different `GSObject`.  (I'm not sure we really need the bit of `GSObject.__init__` that remains either (passing in an `SBProfile`), but that at least wasn't in the way so I just left it).

I decided not to go with a standard python `@property` for `GSObject.noise`, since that would have required a `noise.setter` method and `_noise` instance variable too (to enable users to attach their favorite `_BaseCorrelatedNoise` object to, e.g., `gaussian_galaxy.noise`).  I did find, however, that a [`lazy_property`](http://stackoverflow.com/a/6849299) would work.  It turns out that not only does this delay and cache the computation of an "attribute", due to minutiae of the python descriptor protocol, the cached value is even faster to access than a regular `@property` - I think just as fast as a regular attribute.

With this in place, I removed all the noise propagation bits from the transformation methods and compound constructors, and find that, for example, `sheared = real_galaxy.shear(g1=0.1)` goes down from ~40 microseconds to ~20 microseconds.  Fun :).

Finally, to squash a bug, I decided to have `RandomWalk` inherit from `GSObject` instead of `Sum`.  Despite being a sum of a bunch of Gaussians, `RandomWalk` doesn't really behave like a `Sum`.  In particular, there's no `self._obj_list` attribute, which some of the `Sum` methods depend on.  I think it's simpler to change the parent class than to keep overriding methods that work for `Sum` but not `RandomWalk`.

next up, chromatic noise...